src/asset/asset-file.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @private
+ * @class
+ * @name AssetFile
+ * @classdesc Wraps a source of asset data.
+ */
+class AssetFile {
+    constructor(url, filename, hash, size, opt, contents) {
+        this.url = url || '';
+        this.filename = filename || '';
+        this.hash = hash || null;
+        this.size = size || null;
+        this.opt = opt || null;
+        this.contents = contents || null;
```

Comment:
Maybe consider default parameters here?

Reply:
That wouldn't have the same result. The call site specifies all the constructor's arguments, but some of them may be missing/undefined.

Reply:
I'm not sure I follow, can you give me an example where it would break?
Also, if you passed in 0 for hash or size (yeah, I'm not sure that exactly makes sense but hey)...you'd get null for both, not 0. Not sure what opt is though or what type it is - but same could apply there.

Reply:
I'm not sure if it's better either, but changed hash/size/opt to keep 0's.

For an example, https://github.com/playcanvas/engine/pull/3381/files#diff-59cbf44184755e9db171fb7873b5b92241e4bc05bffd75d001d9e6f738c38700R363 can pass in an undefined argument, which this code deals with, but default arguments wouldn't.

Reply:
OK, cool, thanks.